src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1356,3 +1358,33 @@ def should_encrypt_token_cache(cli_ctx):
     encrypt = cli_ctx.config.getboolean('core', 'encrypt_token_cache', fallback=fallback)
 
     return encrypt
+
+
+def subprocess_arg_type_check(args, kwargs):
+    skip_arg_type_check = kwargs.get("skip_arg_type_check", None)
+    if skip_arg_type_check is not None:
+        del kwargs["skip_arg_type_check"]
+    if skip_arg_type_check:
+        return
+    if isinstance(args, list):
+        return
+    from azure.cli.core.azclierror import ArgumentUsageError
+    raise ArgumentUsageError("cli_subprocess args should be a list of sequence")
+
+
+def subprocess_arg_mask(args, kwargs):
+    enable_arg_mask = kwargs.get("enable_arg_mask", None)
+    if enable_arg_mask is not None:
+        del kwargs["enable_arg_mask"]
+    if not enable_arg_mask:
+        return
+    if not isinstance(args, list):
+        return
+    for i, val in enumerate(args):
+        args[i] = re.sub(ARG_MASK_CHAR, "", val)
```

Comment:
Replacing `ARG_MASK_CHAR` is fragile. Consider using `shlex.quote`: https://docs.python.org/3/library/shlex.html#shlex.quote

Reply:
As the shlex doc said,  "The shlex module is only designed for Unix shells. ", so I planned to put it in cli subprocess doc later for guiding users to split and quote their cmd and args themselves according to the platform.

shell=False and arg list is safe enough for avoid cmd injection.